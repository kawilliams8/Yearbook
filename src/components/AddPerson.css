.AddPerson {
  background-color: rgba(255, 255, 255, 0.95);
  border-radius: 6px;
  margin: auto;
  width: 40%;
}

.formTitle {
  color: blue;
  padding: 10px;
  display: block;
  font-size: 1.5em;
}

button,
input {
  color: red;
  font-size: 1.75em;
  height: 50px;
  margin: 12px;
  width: 350px;
}