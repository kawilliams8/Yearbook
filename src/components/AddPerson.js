import React, {Component} from 'react';
import './AddPerson.css';

class AddPerson extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      quote: "",
      superlative: ""
    }
  }

  handleChange = (event) => {
    this.setState({
      [event.target.name] : event.target.value
    })
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const newStudent = {
      id: Date.now(),
      ...this.state
    }
    this.props.addStudent(newStudent);
    this.setState({
      name: "",
      quote: "",
      superlative: ""
    })
  }
  
  render() {
    return (
      <form className="AddPerson">
      <h3 className="formTitle">Add a New Student</h3>
        <input type="text" name="name" className="nameInput" placeholder="Name" value={this.state.name} onChange={this.handleChange}/>
        <br/>
        <input type="text" name="quote" className="quoteInput" placeholder="Quote" value={this.state.quote} onChange={this.handleChange}/>
        <br/>
        <input type="text" name="superlative" className="superlativeInput" placeholder="Superlative" value={this.state.superlative} onChange={this.handleChange}/>
        <br/>
        <button onClick={this.handleSubmit}>Submit</button>
      </form>
    )
  }
}

export default AddPerson;