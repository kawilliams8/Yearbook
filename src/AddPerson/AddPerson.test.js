import React from 'react';
import { shallow } from 'enzyme';
import AddPerson from './AddPerson';

describe('Add Person', () => {

  let mockAddStudent, mockEvent, wrapper;

  beforeEach(() => {
    mockAddStudent = jest.fn();
    mockEvent = { target: { name: 'name', value: 'MockName' } }
    wrapper = shallow(<AddPerson
      addStudent={mockAddStudent} 
      />)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state information when inputs are changed', () => {
      wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('name')).toEqual('MockName');
  });

  it('should add a student on button click', () => {
    wrapper.find('.AddPerson-button').simulate('click', { preventDefault: () => { } });
    expect(mockAddStudent).toBeCalled();
  });

  it('should clear the inputs after a new student is submitted', () => {
    wrapper.setState({ name: 'MockName', quote: 'MockQuote', superlative: 'MockSuperlative' });
    const expected = { name: '', quote: '', superlative: ''};
    wrapper.instance().clearForm();
    expect(wrapper.state()).toEqual(expected);
  });

});