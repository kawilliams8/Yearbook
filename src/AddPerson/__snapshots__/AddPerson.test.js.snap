// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Person should match the snapshot with all data passed in correctly 1`] = `
<form
  className="AddPerson"
>
  <h3
    className="formTitle"
  >
    Add a New Student
  </h3>
  <input
    className="nameInput"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <br />
  <input
    className="quoteInput"
    name="quote"
    onChange={[Function]}
    placeholder="Quote"
    type="text"
    value=""
  />
  <br />
  <input
    className="superlativeInput"
    name="superlative"
    onChange={[Function]}
    placeholder="Superlative"
    type="text"
    value=""
  />
  <br />
  <button
    className="AddPerson-button"
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
